### PR TITLE
Updated coderabbit rules

### PR DESCRIPTION
**Pull request by krofax**
**Description**
Updated CodeRabbit rules to:

1. Proper Noun Preservation: Proper nouns like "Optimism", "OP Mainnet", and "Ethereum" will now retain their capitalization in headers (H1, H2, H3), even with sentence case.

2. Manual Flagging for Edge Cases: Any headers that don’t follow these rules exactly will be flagged for manual review to avoid false positives.

**Tests**

**Additional context**

**Metadata**

### CODE REVIEW

File: .coderabbit.yaml [hunk 1]
```diff
@@ -14,10 +14,14 @@ reviews:
...
         - Apply the Oxford comma.
-        - Use proper title case for headers, buttons, tab names, page names, and links. Sentence case should be used for body content and short phrases, even in links.
+        - Use proper title case for buttons, tab names, page names, and links. Sentence case should be used for body content and short phrases, even in links.
         - Use correct spelling and grammar at all times (IMPORTANT).
-        - Use sentence case for H1, H2, and H3 headers, capitalizing only the first word and any proper nouns.
-        - For all headers (H1, H2, H3), do not change the capitalization of proper nouns; keep them as they are.
+        - For H1, H2, and H3 headers:
+          1. Use sentence case, capitalizing only the first word.
+          2. Preserve the capitalization of proper nouns, technical terms, and acronyms as defined in the 'nouns.txt' file located in the root directory of the project.
+          3. Do not automatically lowercase words that appear in the 'nouns.txt' file, regardless of their position in the header.
```
Comment (bradleycamacho) [inline .coderabbit.yaml:22]: I'm slightly confused here. So this is saying that CodeRAbbit should look at `nouns.txt` as a whitelist of sorts for words that are allowed to be capitalized in headers correct?

If so, I think we need to take another pass at the list. I think it should exclusively be proper nouns and not just web3 terms

File: nouns.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+Layer 2
+L2
+Rollup
+Smart Contract
+Blockchain
+Cryptocurrency
+Token
+Wallet
+dApp
+Gas Fee
```
Comment (bradleycamacho) [inline nouns.txt:19]:
```suggestion
dApp
```

File: nouns.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Optimism
+OP Mainnet
+Ethereum
+JavaScript
+NFT
+Web3
+DeFi
+DAO
+EVM
+Layer 2
```
Comment (bradleycamacho) [inline nouns.txt:10]:
```suggestion

```

File: .coderabbit.yaml (hunk 1 above)
Comment (krofax) [reply .coderabbit.yaml:22]: Yeah, so this list is by no means a standard, we can always update it, what we have there are placeholders.

Comment (cpengilly) [reply .coderabbit.yaml:22]: agree with both sentiments here, 

- seems like we get the rule and the nouns.txt file in place and we can add to it as we go along (just like we did/have been doing with the words.txt file). 
- but also wonder if we could have coderabbit do an initial sweep thru the docs to populate the current list more?

File: nouns.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+OP Mainnet
+Ethereum
+JavaScript
+NFT
+Web3
+DeFi
+DAO
+EVM
+L2
+dApp
```
Comment (cpengilly) [inline nouns.txt:11]: @krofax we no longer use `dApp` here at OP Labs, a decision from product team (before you two joined). all instances were removed from docs a few months ago and replaced with just `app`

seems like we could add OP Stack, Superchain, Collective, Foundation to this list

**Review by bradleycamacho — APPROVED** [review]

**Comment by cpengilly** [comment]
@krofax confirmed with product that we should use [this file](https://docs.google.com/spreadsheets/d/1iR0QneiiVKe2q0VyoyYBowN89J6Lm0DVwaZ-18SLl1k/edit?gid=752550978#gid=752550978) as source of truth for all product names. perhaps we can write a script to import them directly? not sure if there's a way to automate it.  ❤️

File: nouns.txt (hunk 4 above)
Comment (sbvegan) [inline nouns.txt:11]:
```suggestion
```

Comment (sbvegan) [inline nouns.txt:11]:
```suggestion
OP Stack
```

**Review by cpengilly — APPROVED** [review]
